google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -554,8 +555,20 @@ QueryOptions buildQueryOptions(QueryOptions requestOptions) {
     return builder.build();
   }
 
+  RequestOptions buildRequestOptions(Options options, String txnTag) {
```

Comment:
The transaction tag is already present in the `Options` in the `TransactionContext`, so it would be easier to just keep it there and let the `TransactionContext` deliver it when needed through a `getTransactionTag()` method. See my suggestions below for how that would work out. It would automatically also fix the `TODO` for adding it to read requests.

```suggestion
  RequestOptions buildRequestOptions(Options options) {
```